### PR TITLE
Fix coordinate frames on state message

### LINKED ISSUE #11: LRAUVState fields population status
This ticket keeps track of which fields in `LRAUVState` are populated, to eventually populate all of them. The `LRAUVState` is meant to be a mirror of `SimResultStruct` in the MBARI code base, which contains all the information MBARI’s original simulation provides to the control loop.

LRAUVState is defined in [`lrauv/lrauv_ignition_plugins/proto/lrauv_state.proto`](https://github.com/osrf/lrauv/blob/main/lrauv_ignition_plugins/proto/lrauv_state.proto).

Checked box indicates populated.
Empty box indicates needs work.
(*) indicates populated but need modification (see section below).

- [x] ignition.msgs.Header header
- [ ] int32 errorPad_
- [ ] int32 utmZone_
- [ ] int32 northernHemi_
- [x] float propOmega_
- [ ] float propThrust_
- [ ] float propTorque_
- [x] float rudderAngle_
- [x] float elevatorAngle_
- [x] float massPosition_
- [x] float buoyancyPosition_
- [x] float depth_
<br>

- [x] ignition.msgs.Vector3d rph_  // roll_, pitch_, heading_ in SimResultStruct
<br>

- [x] float speed_
- [ ] double latitudeDeg_(#90)
- [ ] double longitudeDeg_ (#90)
- [ ] float netBuoy_
<br>

- [ ] ignition.msgs.Vector3d force_  // forceX_, forceY_, forceZ_ in SimResultStruct
- [x] ignition.msgs.Vector3d pos_  // posX_, posY_, posZ_ in SimResultStruct (#48)
- [x] ignition.msgs.Vector3d posRPH_  // posRoll_, posPitch_, posHeading_ in SimResultStruct
- [x] ignition.msgs.Vector3d posDot_  // posXDot_, posYDot_, posZDot_ in SimResultStruct (#48) 
- [ ] ignition.msgs.Vector3d rateUVW_  // rateU_, rateV_, rateW_ in SimResultStruct
- [ ] ignition.msgs.Vector3d ratePQR_  // rateP_, rateQ_, rateR_ in SimResultStruct
<br>

- [x] float northCurrent_ (https://github.com/osrf/lrauv/pull/77)
- [x] float eastCurrent_ (https://github.com/osrf/lrauv/pull/77)
- [x] float vertCurrent_ (https://github.com/osrf/lrauv/pull/77)
- [ ] float magneticVariation_
- [ ] float soundSpeed_
- [x] float temperature_ (https://github.com/osrf/lrauv/pull/77)
- [x] float salinity_ (https://github.com/osrf/lrauv/pull/77)
- [ ] float density_
- [ ] repeated float values_  // Size 4 (https://github.com/osrf/lrauv/pull/77)

### Populated but need modification:

Note that the coordinate frame convention interpreted by the MBARI controller is FSK.

- [x] depth_ and rph_ are currently in the ROS coordinate frame convention - x forward, z up, y left. Need to change to the FSK convention that MBARI's controller uses - x fore, y starboard (right), z keel (down). (#48)

@braanan @arjo129

### PR DESCRIPTION
**Pull request by chapulina**
* Wrap up #48 
* Closes #11 
* #104 helps understanding this PR, but it's not required
* Make sure all fields are documented
* Test all populated fields on state message except for sensor data (wait for interpolation)
* Update published message to make sure quantities are in the documented frames.

### CODE REVIEW

**Comment by arjo129** [comment]
Hey, @chapulina  I've not had time to look into this. But on my PC when I run the tests I'm also seeing segfaulting and regression in a lot of the integration test cases. It seems to cause the depth_pitchmass_vbs test to fail consistently...

**Comment by chapulina** [comment]
I narrowed down the controller failures to the RPH orientation. I'm still trying to understand the expectations around those values, see comments on 512bd2f.

File: lrauv_ignition_plugins/src/TethysCommPlugin.cc [hunk 1]
```diff
@@ -546,56 +552,78 @@ void TethysCommPlugin::PostUpdate(
...
+      this->initialModelPoseNED.value().Inverse();
+  ignition::msgs::Set(stateMsg.mutable_pos_(), poseOffsetNED.Pos());
+  // ignition::msgs::Set(stateMsg.mutable_rph_(), poseOffsetNED.Rot().Euler());
+  // ignition::msgs::Set(stateMsg.mutable_posrph_(), poseOffsetNED.Rot().Euler());
+
+  // TODO(chapulina) The controller fails with the orientation above. Example error:
+  //
+  // [VerticalControl](CRITICAL): Excessive depth excursion=10.573917 m, failToGoUpDepth_=17.503380 m, depthRate=0.211946 m/s, pitch =0.114867 deg.
+  //
+  // The difference between poseOffsetNED.Rot() and rph is that roll and pitch are swapped.
```
[the remark this answers is not in the document]
Comment (chapulina) [reply lrauv_ignition_plugins/src/TethysCommPlugin.cc:579]: I just verified that if the vehicle starts facing North, the NED orientation works for the controllers.

[the remark this answers is not in the document]
Comment (chapulina) [reply lrauv_ignition_plugins/src/TethysCommPlugin.cc]: Pushed in 23e7631, checking with @braanan before updating all test expectations.

File: lrauv_ignition_plugins/proto/lrauv_state.proto [hunk 2]
```diff
@@ -64,35 +83,62 @@ message LRAUVState
...
-  float density_                  = 31;  // Density of the surrounding water in Kg / m ^ 3
-  repeated float values_          = 32;  // Size 4 (0: chlorophyll in ug / L, 1: pressure in Pa)
+
+  ignition.msgs.Vector3d pos_     = 19;  // Vehicle's offset position in "world frame"
+                                         // (see above) with respect to its initial
+                                         // position. Unit: meters
+
+  ignition.msgs.Vector3d posRPH_  = 20;  // Vehicle's offset orientation in "world frame"
+                                         // (see above) with respect to its initial
+                                         // orientation. Unit: rad
```
Comment (chapulina) [inline lrauv_ignition_plugins/proto/lrauv_state.proto:110]: Just talked to @braanan , and these shouldn't be offsets with respect to the vehicle's initial position / orientation, but with respect to the NED world origin. That is, if the vehicle starts facing West, then it has a 270 degree heading from the beginning.

The same goes for all vehicles, even those spawned after the first one. They all use the same reference world frame.

**Comment by chapulina** [comment]
Moving this PR to draft. I'll be breaking some of the smaller changes into new PRs, and update this PR to reorient the model in a way that it faces North at zero orientation. While at it, I'll also tackle #80 .

**Comment by chapulina** [comment]
Closing this PR, it's being broken into smaller ones, starting with:

* #143 
* #154 
* #155

**Pull request closed without merging**